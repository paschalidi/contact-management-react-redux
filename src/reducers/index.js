import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import reducerAddDeleteContact from './reducer-add-delete-contact'

const rootReducer = combineReducers({
  form: formReducer,
  reducerAddDeleteContact
});

export default rootReducer;
