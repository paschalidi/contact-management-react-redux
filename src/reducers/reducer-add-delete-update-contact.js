import {
  ADD_UPDATE_CONTACT,
  DELETE_CONTACT,
} from '../actions/types'

const INITIAL_STATE = {
  0: {
    id: null,
    firstName: null,
    lastName: null,
    email: null
  }
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case ADD_UPDATE_CONTACT:
      const { id, firstName, lastName, email } = action.payload;

      return Object.assign(
        {},
        state,
        { [id]: { id, firstName, lastName, email } }
      );
    case DELETE_CONTACT:
      const { identifier } = action.payload;
      let newState = Object.assign({}, state);

      delete newState[identifier];
      return newState;

    default:
      return state;
  }
}