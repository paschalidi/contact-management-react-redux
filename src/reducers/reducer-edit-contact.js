import {
  EDIT_CONTACT
} from '../actions/types'

const INITIAL_STATE = {
  id: null,
  firstName: null,
  lastName: null,
  email: null
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case EDIT_CONTACT:
      return {
        data: action.payload
      };
    default:
      return state;
  }
}