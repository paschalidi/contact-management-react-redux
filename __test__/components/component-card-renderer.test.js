import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

import reducers from '../../src/reducers';
import CardRenderer from '../../src/components/component-card-renderer'

import renderer from 'react-test-renderer';

describe('CardRenderer', () => {
  const createStoreWithMiddleware = applyMiddleware()(createStore);
  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={createStoreWithMiddleware(reducers)}>
        <CardRenderer />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});