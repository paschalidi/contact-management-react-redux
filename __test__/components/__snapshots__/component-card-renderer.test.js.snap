// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRenderer renders correctly 1`] = `
<div
  className="container-show-contact"
>
  <div
    className="row"
  />
</div>
`;

exports[`CardRenderer should dispatch action when clicking like 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowContacts)
    contacts={
      Object {
        "1": Object {
          "email": "mail",
          "firstName": "name",
          "id": 1,
          "lastName": "lastname",
        },
        "2": Object {
          "email": "mail",
          "firstName": "name",
          "id": 1,
          "lastName": "lastname",
        },
        "3": Object {
          "email": "mail",
          "firstName": "name",
          "id": 1,
          "lastName": "lastname",
        },
      }
    }
  >
    <ShowContacts
      contacts={
        Object {
          "0": Object {
            "email": null,
            "firstName": null,
            "id": null,
            "lastName": null,
          },
        }
      }
      dispatch={[Function]}
    >
      <div
        className="container-show-contact"
      >
        <div
          className="row"
        />
      </div>
    </ShowContacts>
  </Connect(ShowContacts)>
</Provider>
`;
